Automattic/woocommerce-payments
# Add basic usage tracking in deposits list

Fixes #5884

#### Changes proposed in this Pull Request

This PR updates basic tracking in the deposits list view. 

The only new event is:

- `wcadmin_wcpay_deposits_row_click`, fired when merchant clicks a deposit row to view detail.

The existing events are relevant and unchanged:

- `wcadmin_page_view path=payments_deposits` fired on page view.
- `wcadmin_wcpay_deposits_download`, fired when merchant downloads a deposits export.
   - Note in this PR I've moved this event name to a constant in `wcpayTracks` for consistency.


<img width="2012" alt="Screenshot 2023-05-05 at 3 40 18 PM" src="https://user-images.githubusercontent.com/4167300/236373424-5d63e969-9b94-42f8-ba0f-b427d648c664.png">

#### Testing instructions

- Onboard WCPay and set up a site with a balance, and some existing deposits. 
- Ensure usage tracking is enabled: `WooCommerce > Settings > Advanced > WooCommerce.com > Usage tracking`. See below for info on viewing individual events in browser.
- Go to `Payments > Deposits`.
  - Confirm view event fires: `wcadmin_page_view` with `path=payments_deposits`
- Click `☁️ Download`.
  - Confirm download event fires: `wcadmin_wcpay_deposits_download`
- Click a deposit row to view details.
  - Confirm row event fires: `wcadmin_wcpay_deposits_row_click`

##### How to view tracks events
These instructions are for Google Chrome, might be different in other browsers.

- Open browser `Developer tools > Network` tab so you can see network requests.
- Search for `t.gif` (this is the endpoint for tracks events).
- Each request is an event.
- Click the request and view `Payload` to see params.
  - `_en` is the event name.

-------------------


**Post merge**



- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : n/a, usage tracking does not need QA testing
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. n/a, usage tracking does not change critical flows
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. n/a, internal-facing feature only
